X86/X86MachineInstructionRaiserUtils.cpp

```diff
@@ -335,6 +359,39 @@ bool X86MachineInstructionRaiser::recordDefsToPromote(unsigned PhysReg,
   return true;
 }
 
+// Find the MachineInstr in MBB where PhysReg is used, return nullptr otherwise
+const MachineInstr *X86MachineInstructionRaiser::getInstUsingPhysRegInBlock(
```

Comment:
This function appears to be near identical to `hasPhysRegDefInBlock()`. To avoid duplicating functionality they should be merged. Attached is a suggested patch. Please make the necessary changes to the call sites of `hasPhysRegDefInBlock()` by testing for the returned value to be non-null.
[PR-86-suggested-2.txt](https://github.com/microsoft/llvm-mctoll/files/6524717/PR-86-suggested-2.txt)

Reply:
Thanks for the patch!